lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -510,7 +510,14 @@ public TokenDetails authorize(TokenParams params, AuthOptions options) throws Ab
 			tokenDetails = authOptions.tokenDetails;
 			setTokenDetails(tokenDetails);
 		} else {
-			tokenDetails = assertValidToken(params, options, true);
+			try {
+				tokenDetails = assertValidToken(params, options, true);
+			} catch (AblyException e) {
+				/* Give AblyRealtime a chance to update its state and emit an event according to RSA4c */
+				if (e.errorInfo.code == 80019)
```

Comment:
I don't see why `onAuthError()` is only called if there was an error from `authCallback` or `authUrl`; there are other reasons for an `authorise()` failing; eg a token request failed because of a rate limit. The `ConnectionManager` will need to be notified of that problem as well.